### PR TITLE
Fix ``used-before-assignment`` for conditional self-referential typing

### LINKED ISSUE #5499: Regression `used-before-assignment` with type annotation in conditional
### Bug description

```py
import random

class A:
    def f(self) -> None:
        if random.random() < 0.5:
            _x: A = self  # false-positive: used-before-assignment

    def g(self) -> None:
        _x: A = self  # no warning
```

This is the same issue reported and fixed in #5342 but it still occurs if the assignment is inside a conditional.

### Configuration

_No response_

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:6:16: E0601: Using variable 'A' before assignment (used-before-assignment)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.6 (default, Aug 16 2021, 15:13:47) 
[GCC 9.3.0]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #5499.

I should have thought of this fix in #5342. This is much better..

### CODE REVIEW

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -1805,7 +1805,7 @@ def _is_first_level_self_reference(
         """
         if (
             node.frame().parent == defstmt
-            and node.statement(future=True) not in node.frame().body
+            and node.statement(future=True) == node.frame()
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/variables.py:1808]: I don't understand how this work, could you explain ?

Comment (DanielNoord) [reply pylint/checkers/variables.py:1808]: Basically for arguments in function definitions their statement is also their frame (the function definition). For nodes within the body or even within sub-bodies (conditionals) the statement will be different from the frame (the latter still being the function definition).

Comment (Pierre-Sassoulas) [reply pylint/checkers/variables.py:1808]: That makes sense, thank you.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**